algorithms/integration/kapton_2019_correction.py

```diff
@@ -0,0 +1,538 @@
+            edge_idx=(edge_idx[1],edge_idx[0],edge_idx[2],edge_idx[3])
+        if v2 < 0.0:
+            edge_idx=(edge_idx[0],edge_idx[1],edge_idx[3],edge_idx[2])
+
+
+        # Now make sure the edges and the kapton lines are on the right side (i.e not swapped).
+        # Let's look at edge_idx[0:2] i,e the first edge of detector parallel to the kapton
+        pt1=edge_pts[edge_idx[0]]
+        pt2=edge_pts[edge_idx[1]]
+        # Now find the distance between each of these points and the kapton lines.
+        d1_kapton_1=self.distance_of_point_from_line(pt1, int_edge_pts[0], int_edge_pts[1])
+        d1_kapton_2=self.distance_of_point_from_line(pt1, int_edge_pts[2], int_edge_pts[3])
+        d2_kapton_1=self.distance_of_point_from_line(pt2, int_edge_pts[0], int_edge_pts[1])
+        d2_kapton_2=self.distance_of_point_from_line(pt2, int_edge_pts[2], int_edge_pts[3])
+        if d1_kapton_1 < d1_kapton_2: # closer to max than edge
+            assert d2_kapton_1 < d2_kapton_2, 'Distance mismatch. Edge of detector might be on wrong side of kapton tape ... please check'
+            pair_values = [(edge_pts[edge_idx[0]],edge_pts[edge_idx[1]],edge_pts[edge_idx[2]], edge_pts[edge_idx[3]]),
+                                 (int_edge_pts[0],int_edge_pts[1], int_edge_pts[2], int_edge_pts[3])]
+        else:
+            pair_values = [(edge_pts[edge_idx[0]],edge_pts[edge_idx[1]],edge_pts[edge_idx[2]], edge_pts[edge_idx[3]]),
+                                 (int_edge_pts[2],int_edge_pts[3], int_edge_pts[0], int_edge_pts[1])]
+
+        return pair_values
+
+    def abs_bounding_lines_on_image(self, detector):
+        pair_values=self.abs_bounding_lines_in_mm(detector)
+        r0,r1,r2,r3=pair_values[0]
+        ra,rb,rc,rd=pair_values[1]
+        # Get slope, intercept from 0-2 edge line and 1-3 edge line
+        def get_line_equation(rA, rB):
+            xA, yA, zA = rA
+            xB, yB, zB = rB
+            m = (yB-yA)/(xB-xA)
+            c = yA-m*xA
+            return (m,c)
+        def get_line_intersection(m1,c1,m2,c2):
+            x = (c2-c1)/(m1-m2)
+            y = m1*x+c1
+            return (x,y)
+
+        m03,c03=get_line_equation(r0,r3)
+        m12,c12=get_line_equation(r1,r2)
+
+        # Get slope, intercept for a-b and c-d
+        mab,cab=get_line_equation(ra,rb)
+        mcd,ccd=get_line_equation(rc,rd)
+        # Get intersection between 0-2 with a-b
+        r03_ab=get_line_intersection(m03,c03,mab,cab)
+        # Get line intersection between 1-3 and a-b
+        r12_ab=get_line_intersection(m12,c12,mab,cab)
+        # Get line intersection between 0-2 and c-d
+        r03_cd=get_line_intersection(m03,c03,mcd,ccd)
+        # Get line intersection between 1-3 and c-d
+        r12_cd=get_line_intersection(m12,c12,mcd,ccd)
+        # returns pairs of x,y
+        return [(r03_ab[0], r03_ab[1], r12_ab[0], r12_ab[1]), (r03_cd[0], r03_cd[1], r12_cd[0], r12_cd[1])]
+
+class image_kapton_correction(object):
```

Comment:
Less important, but something we've been trying to catch ourselves doing; This class appears to exist only to:
```python
some_action = image_kapton_correction(param, param, configuration, configuration)
result = some_action()
```
this isn't a class, it's a function. Is there any reason why it can't just be `result = image_kapton_correction(....)` We're actively trying to slowly refactor this pattern our where unnecessary.